src/classes/video_data.py

```diff
@@ -1,8 +1,76 @@
-from typing import Any
+# MIT License
+#
+# Copyright (c) 2023 Victor Calderon and Travis Craft
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
 
-from pydantic import BaseModel
+import logging
 
+import validators
 
-class VideoData(BaseModel):
-    raw_video: Any
-    url: str
+__author__ = ["Victor Calderon and Travis Craft"]
+__maintainer__ = ["Victor Calderon and Travis Craft"]
+__all__ = ["VideoDataDAO"]
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+logger.setLevel(logging.INFO)
+
+
+class VideoDataDAO(object):
```

Comment:
I see you swapped out the Pydantic `BaseModel` for `object` here. Did you run into any issues with the `BaseModel`? Honestly I'm still learning a lot of best practices in Python so I'm good either way but wanted to check in on the swap.

And just a small nitpick but personally I try to avoid encoding things like DAO into class names. I'm reading this as Data Access Object which I typically associate with database models. I think this is more inline with Data Transfer Object or POJOs (Plain Old Java Objects).

Reply:
Yes, thanks for pointing that out. I updated this since it had given me an error during testing, but it was probably that I hadn't spent much time using `BaseModel`. I haven't used it before, but I'm down to keeping it if you think it's better. In the past, I've used `object`, but I'm happy to keep `BaseModel` if you think there are benefits to it.

Regarding the `DAO` as part of the name, I've reverted it to its original name, i.e. `VideoData`, since I'm not really attached to the name.